lib/executor/executor.cpp

```diff
@@ -206,5 +207,190 @@ Executor::asyncInvoke(const Runtime::Instance::FunctionInstance *FuncInst,
           std::vector(ParamTypes.begin(), ParamTypes.end())};
 }
 
+void Executor::generateCoredump(Runtime::StackManager &StackMgr) {
+  const Configure Conf;
+  Loader::Serializer Ser(Conf);
+  const auto *CurrentInstance = StackMgr.getModule();
+  if (CurrentInstance == nullptr) {
+    return;
+  }
+
+  auto Core = collectProcessInformation();
+  // auto DataSection = collectDataSection(StackMgr);
+  auto MemSec = Executor::collectMemory(StackMgr);
+  // auto Globals = collectGlobals(StackMgr);
+  auto CoreStack = collectCoreStack(StackMgr);
+  // auto CoreInstances = collectCoreInstances(StackMgr);
+  // auto CoreModules = collectCoreModules(StackMgr);
+
+  //  final file generation
+  AST::Module Mod{};
+  std::vector<Byte> &Magic = Mod.getMagic();
+  Magic.insert(Magic.begin(), {0x00, 0x61, 0x73, 0x6d});
+
+  std::vector<Byte> &Version = Mod.getVersion();
+  Version.insert(Version.begin(), {0x01, 0x00, 0x00, 0x00});
+
+  Mod.getCustomSections().emplace_back(Core);
+  // Mod.getCustomSections().emplace_back(CoreModules);
+  // Mod.getCustomSections().emplace_back(CoreInstances);
+  Mod.getCustomSections().emplace_back(CoreStack);
+  // Mod.getDataSection() = std::move(DataSection);
+  Mod.getMemorySection() = std::move(MemSec);
+  // TODO globals works with DWARF data, is this section needed?
```

Comment:
I still don't know why the `Globals` is related to `DWARF data`. But we must store the globals section.

Reply:
I've tried with using the globals section and without it. For the examples I've run for some reason `info globals` still works(the same is true for coredump-rewriter, it does not include a globals section). Regardless I'll include the globals section to store it.